src/libertem/udf/stddev.py

```diff
@@ -0,0 +1,189 @@
+import collections
+
+import numpy as np
+
+from libertem.common.buffers import BufferWrapper
+
+
+VariancePart = collections.namedtuple('VariancePart', ['var', 'sum_im', 'N'])
+
+
+def batch_buffer():
+    """
+    Initializes BufferWrapper objects for sum of variances,
+    sum of frames, and the number of frames
+
+    Returns
+    -------
+    A dictionary that maps 'var', 'std', 'mean', 'num_frame', 'sum_frame' to
+    the corresponding BufferWrapper objects
+    """
+    return {
+        'var': BufferWrapper(
+            kind='sig', dtype='float32'
+            ),
+        'std': BufferWrapper(
+            kind='sig', dtype='float32'
+            ),
+        'mean': BufferWrapper(
+            kind='sig', dtype='float32'
+            ),
+        'num_frame': BufferWrapper(
+            kind='single', dtype='float32'
+            ),
+        'sum_frame': BufferWrapper(
+            kind='sig', dtype='float32'
+            )
+    }
+
+
+def compute_batch(frame, var, std, mean, sum_frame, num_frame):
+    """
+    Given a frame, update sum of variances, sum of frames,
+    and the number of total frames
+
+    Parameters
+    ----------
+    frame
+        single frame of the data
+
+    var
+        Buffer that stores sum of variances of the previous set of frames
+
+    sum_frame
+        Buffer that sores sum of frames of the previous set of frames
+
+    num_frame
+        Buffer that stores the number of frames used for computation
+
+    """
+    if num_frame == 0:
+        var[:] = 0
+
+    else:
+        p0 = VariancePart(var=var, sum_im=sum_frame, N=num_frame)
+        p1 = VariancePart(var=0, sum_im=frame, N=1)
+        compute_merge = merge(p0, p1)
+
+        var[:] = compute_merge.var
+
+    sum_frame[:] += frame
+    num_frame[:] += 1
+
+
+def batch_merge(dest, src):
+    """
+    Given two buffers that contain sum of variances, sum of frames,
```

Comment:
`dest` and `src` aren't buffers' I'd rather describe them as partial results (same below in the Parameters part)